### PR TITLE
cmd: add xdev tool

### PR DESCRIPTION
**Pull request by icexin**
## Description

Add command line tools for build and test contract project

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Brief of your solution

A new tool called `xdev`.

- `xdev build` to build a contract package
- `xdev test` to test a contract package

## How Has This Been Tested?

Add go test and self test

### CODE REVIEW

File: core/cmd/xdev/internal/cmd/test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const (
+	testDir = "test"
+)
+
+func (c *testCommand) testPackage() error {
+	root, err := findPackageRoot()
+	if err != nil {
+		return err
+	}
+	os.Chdir(root)
```
Comment (fxsjy) [inline core/cmd/xdev/internal/cmd/test.go]: err handling?

Comment (icexin) [reply core/cmd/xdev/internal/cmd/test.go]: done

File: core/cmd/xdev/internal/cmd/build.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	if err != nil {
+		return err
+	}
+	c.builder = b
+	c.entryPkg = pkg
+	return nil
+}
+
+func (c *buildCommand) xchainRoot() (string, error) {
+	xroot := os.Getenv("XROOT")
```
Comment (yucaowang) [inline core/cmd/xdev/internal/cmd/build.go]: We have a XCHAIN_ROOT, is this the same with XCHAIN_ROOT?

Comment (icexin) [reply core/cmd/xdev/internal/cmd/build.go]: Yes, new code has reused XCHAIN_ROOT.

**Pull request merged**